js/game.js

```diff
@@ -30,31 +34,92 @@ export default class Game {
         this.tamagotchi.health.value--;
       this.tamagotchi.displayHealth(this.tamagotchi.health.element);
     }
+    this.tamagotchi.updateState();
   };
 
   decreaseHunger = () => {
+    this.tamagotchi.checkState();
     if (this.tamagotchi.hunger.value > 0) this.tamagotchi.hunger.value--;
     this.tamagotchi.displayHunger(this.tamagotchi.hunger.element);
+    this.tamagotchi.updateState();
   };
 
   decreaseEnergy = () => {
+    this.tamagotchi.checkState();
     if (this.tamagotchi.energy.value > 0) {
       this.tamagotchi.energy.value--;
       if (this.tamagotchi.fun.value <= 0) this.tamagotchi.energy.value--;
     }
     this.tamagotchi.displayEnergy(this.tamagotchi.energy.element);
+    this.tamagotchi.updateState();
   };
 
   decreaseFun = () => {
+    this.tamagotchi.checkState();
     if (this.tamagotchi.fun.value > 0) this.tamagotchi.fun.value--;
     this.tamagotchi.displayFun(this.tamagotchi.fun.element);
+    this.tamagotchi.updateState();
   };
 
   end = () => {
     clearInterval(this.healthIntervalId);
     clearInterval(this.hungerIntervalId);
     clearInterval(this.energyIntervalId);
     clearInterval(this.funIntervalId);
-    console.log("Game over");
+  };
+
+  resume = () => {
+    this.healthIntervalId = setInterval(this.decreaseHealth, 1000);
+    this.hungerIntervalId = setInterval(this.decreaseHunger, 1000);
+    this.energyIntervalId = setInterval(this.decreaseEnergy, 2000);
+    this.funIntervalId = setInterval(this.decreaseFun, 1000);
+  };
+
+  eatBtn = document.getElementById("eatBtn");
+  sleepBtn = document.getElementById("sleepBtn");
+  playBtn = document.getElementById("playBtn");
+  nimo = document.getElementById("nimo");
+  stateText = document.getElementById("petState");
+
+  btnListeners = () => {
+    eatBtn.addEventListener("click", () => {
```

Comment:
**nitpick:** These 3 methods looks really similar. It would be nice to make 1 method that you could parametrize, and reuse it. Example:
```
doAction = ({
  actionClass,
  actionText,
  onAction,
  onActionEffect
}) => {
...
}
```

These are 4 different parameters that I found in your below action functions that are differing them from each other. Do you see them as well?:) Doing it this way you could add more actions without duplicating logic.

Reply:
Probably I still got it wrong, but I made some changes to make it more "parametrized". Please tak a look.